### DISCUSSION TITLE
m3ufu question:  Getting failure on some m3u8 file;  Sorry of the wrong area

### DISCUSSION THREAD

**Discussion by pcb105** [futzu/threefive#93] [Q&A] [answered]
Sorry if this is not the right location for this question....

If I run the following m3u8 input.... It appears to work. 
python3 m3ufu.py -i https://playertest.longtailvideo.com/adaptive/captions/playlist.m3u8

When I run with the following m3u8.  I get errors...
python3 m3ufu.py -i https://d15an60oaeed9r.cloudfront.net/live_stream_v2/sports_reel_with_markers.m3u8
python3 m3ufu.py -i https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8
I am not sure how to go about debugging what the issue(s) are.....

I see that the working one has ts segments (does the sw expect TS files)?
Does the software expect / handle a single segment (audio / video muxed) Does it handle separate AV segments
TIA: Paul

**Comment by futzu** [comment, marked as answer]
you know can just run it like
```lua
 m3ufu  -i https://d15an60oaeed9r.cloudfront.net/live_stream_v2/sports_reel_with_markers.m3u8
```
those are master m3u8 files,  pick one of the renditions to see the segments like:
```lua
m3ufu -i https://d15an60oaeed9r.cloudfront.net/live_stream_v2/sports_reel_with_markers_1080.m3u8
```
## OR
try [showcues](https://github.com/futzu/showcues)
A company named tunein.com hired me to do a variation on m3ufu to make it easier to debug SCTE-35 in HLS. 
It does handle master.m3u8 files
run it like
```lua 
showcues https://d15an60oaeed9r.cloudfront.net/live_stream_v2/sports_reel_with_markers.m3u8
```
![image](https://github.com/futzu/scte35-threefive/assets/52701496/97a21e15-4a5a-4815-89fe-7745d6a17295)

run `showcues help`
to get the particulars.

**Reply by pcb105** [reply]
Thanks for the response.  I am running on a Mac (macOS Sonoma) Python Version 3.9.6. 
When I run ./showcues https://d15an60oaeed9r.cloudfront.net/live_stream_v2/sports_reel_with_markers.m3u8
I get 
```
None
Traceback (most recent call last):
  File "/Users/paulbrown/Documents/projects/m3u8/showcues-main/bin/./showcues", line 7, in <module>
    cli()
  File "/Users/paulbrown/Library/Python/3.9/lib/python/site-packages/showcues.py", line 912, in cli
    fu.decode()
  File "/Users/paulbrown/Library/Python/3.9/lib/python/site-packages/m3ufu.py", line 618, in decode
    if not self._parse_line(line):
  File "/Users/paulbrown/Library/Python/3.9/lib/python/site-packages/m3ufu.py", line 599, in _parse_line
    self._do_media(line)
  File "/Users/paulbrown/Library/Python/3.9/lib/python/site-packages/m3ufu.py", line 565, in _do_media
    self._add_media(media)
  File "/Users/paulbrown/Library/Python/3.9/lib/python/site-packages/m3ufu.py", line 546, in _add_media
    segment.decode()
  File "/Users/paulbrown/Library/Python/3.9/lib/python/site-packages/m3ufu.py", line 400, in decode
    self.end = round(self.start + self.duration, 6)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```

**Reply by pcb105** [reply]
This is the same error I get when I run with m3ufu.  Any ideas?

**Reply by futzu** [reply]
Sorry , I don't always see stuff in the discussions.
It didn't get  a start time for the segment, which is odd, because I'm running showcues on it now with no problems.

![image](https://github.com/futzu/scte35-threefive/assets/52701496/431cd769-aeba-448a-bc58-ec6263df4a54)

It's got to be on your end, I am not to re-create the error, even with your own stream.
maybe make sure you're sure you are using the latest greatest, and pip install them

`python3 -m pip install --upgrade m3ufu showcues`



if you're dumb like me and run debian, then use,
`python3 -m pip install --upgrade m3ufu showcues  --break-system-packages`

I fought hard against pep 668, but they banned from python over it.
